apps/aecore/src/aec_sync.erl

```diff
@@ -110,19 +110,22 @@ handle_worker(Task, Action) ->
 %% to pick a random hash from the hashes in the pool.
 
 -type chain_id() :: reference().
--type chain_block() :: #{ hash := aec_blocks:block_header_hash()
-                        , height := aec_blocks:height()
-                        }.
--type chain() :: #{ chain_id := chain_id()
-                  , peers := [aec_peers:peer_id()]
-                  , chain := [chain_block(), ...]
-                  }.
+-record(chain_block, { hash :: aec_blocks:block_header_hash()
```

Comment:
Records makes the code much easier to follow statically for process state that does not travel across module boundaries.

----

From https://learnyousomeerlang.com/maps

> ... despite the similarity in syntax between maps and records, dicts and maps are much closer together semantically than maps are to records. Therefore, using maps to replace records would be similar to trying to replace 'structs' in a language like C with a hash map. It's not impossible, but that doesn't mean it's a good idea given they often have different purposes.
>
> Keys being known at compile time brings advantages with fast access to specific values (faster than what is possible dynamically), additional safety (crash early rather than corrupting state), and easier type checking. These make records absolutely appropriate for a process' internal state, despite the occasional burden of writing a more verbose code_change function.
>
> On the other hand, where Erlang users would use records to represent complex nested key/value data structures (oddly similar to objects in object-oriented languages) that would frequently cross module boundaries, maps will help a lot. Records were the wrong tool for that job.
>
> To put it briefly, places where records really felt out of place and cumbersome could be replaced by maps, but most record uses should not be replaced that way.